pkg/monitor/intervalcreation/rendering_per_namespace.go

```diff
@@ -148,3 +160,34 @@ func (r podRendering) WriteEventData(artifactDir string, events monitorapi.Inter
 
 	return utilerrors.NewAggregate(errs)
 }
+
+// WriteEventData for ingressServicePodRendering writes out a custom spyglass chart to help debug TRT-364 and BZ2101622 where
+// image-registry, console, and oauth pods were experiencing disruption during upgrades.  We wanted one chart that
+// showed those pods, router-default pods, node changes, and disruption.
+//
+func (r ingressServicePodRendering) WriteEventData(artifactDir string, events monitorapi.Intervals, timeSuffix string) error {
+	errs := []error{}
+	disruptionReasons := sets.NewString(backenddisruption.DisruptionBeganEventReason,
+		backenddisruption.DisruptionEndedEventReason,
+		backenddisruption.DisruptionSamplerOutageBeganEventReason)
+	writer := NewNonSpyglassEventIntervalRenderer("image-reg-console-oauth",
+		func(eventInterval monitorapi.EventInterval) bool {
+			switch {
+			case isInterestingNamespace(eventInterval,
+				sets.NewString("openshift-authentication", "openshift-console", "openshift-image-registry", "openshift-ingress")):
```

Comment:
build the set outside of this function to avoid re-allocating each time.  It's small, but tidy.